osu.Game/Overlays/Profile/Header/Components/DailyChallengeStreakTooltip.cs

```diff
@@ -0,0 +1,251 @@
+
+            // bestDaily.Value = UsersStrings.ShowDailyChallengeUnitDay(statistics.DailyStreakBest.ToLocalisableString(@"N0"));
+            bestDaily.Value = $"{statistics.DailyStreakBest:N0}d";
+            bestDaily.ValueColour = colours.ForRankingTier(TierForDaily(statistics.DailyStreakBest));
+
+            // bestWeekly.Value = UsersStrings.ShowDailyChallengeUnitWeek(statistics.WeeklyStreakBest.ToLocalisableString(@"N0"));
+            bestWeekly.Value = $"{statistics.WeeklyStreakBest:N0}w";
+            bestWeekly.ValueColour = colours.ForRankingTier(TierForWeekly(statistics.WeeklyStreakBest));
+
+            topTen.Value = statistics.Top10PercentPlacements.ToLocalisableString(@"N0");
+            topTen.ValueColour = colourProvider.Content2;
+
+            topFifty.Value = statistics.Top50PercentPlacements.ToLocalisableString(@"N0");
+            topFifty.ValueColour = colourProvider.Content2;
+        }
+
+        // reference: https://github.com/ppy/osu-web/blob/8206e0e91eeea80ccf92f0586561346dd40e085e/resources/js/profile-page/daily-challenge.tsx#L13-L43
+        public static RankingTier TierForDaily(int daily)
+        {
+            if (daily > 360)
+                return RankingTier.Lustrous;
+
+            if (daily > 240)
+                return RankingTier.Radiant;
+
+            if (daily > 120)
+                return RankingTier.Rhodium;
+
+            if (daily > 60)
+                return RankingTier.Platinum;
+
+            if (daily > 30)
+                return RankingTier.Gold;
+
+            if (daily > 10)
+                return RankingTier.Silver;
+
+            if (daily > 5)
+                return RankingTier.Bronze;
+
+            return RankingTier.Iron;
+        }
+
+        public static RankingTier TierForWeekly(int weekly) => TierForDaily((weekly - 1) * 7);
+
+        protected override void PopIn() => this.FadeIn(200, Easing.OutQuint);
+
+        protected override void PopOut() => this.FadeOut(200, Easing.OutQuint);
+
+        public void Move(Vector2 pos) => Position = pos;
+
+        private partial class StreakPiece : FillFlowContainer
+        {
+            private readonly OsuSpriteText valueText;
+
+            public LocalisableString Value
+            {
+                set => valueText.Text = value;
+            }
+
+            public ColourInfo ValueColour
+            {
+                set => valueText.Colour = value;
+            }
+
+            public StreakPiece(LocalisableString title)
+            {
+                AutoSizeAxes = Axes.Both;
+                Direction = FillDirection.Vertical;
+
+                Children = new Drawable[]
+                {
+                    new OsuSpriteText
+                    {
+                        Font = OsuFont.GetFont(size: 12),
+                        Text = title,
+                    },
+                    valueText = new OsuSpriteText
+                    {
+                        // Colour = colour
```

Comment:
?

Reply:
Don't recall what I had in mind here, will remove.